container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go

```diff
@@ -1748,6 +2070,140 @@ func (manager *KubernetesManager) RunExecCommandWithStreamedOutput(
+	// pod needs to be privileged to access host filesystem
+	isPrivileged := true
+	hasHostPidAccess := true
+	hasHostNetworkAccess := true
+	removeDataDirPodName := "remove-dir-pod"
+	hostVolumeName := "host"
+	hostMountPath := "/host"
+	removeDataDirPod, err := manager.CreatePod(ctx,
+		namespace,
+		removeDataDirPodName,
+		nil,
+		nil,
+		nil,
+		[]apiv1.Container{
+			{
+				Name:  removeContainerName,
+				Image: "busybox",
+				Command: []string{
+					"sh",
+					"-c",
+					"sleep 10000000s",
+				},
+				Args:       nil,
+				WorkingDir: "",
+				Ports:      nil,
+				EnvFrom:    nil,
+				Env:        nil,
+				Resources: apiv1.ResourceRequirements{
+					Limits:   nil,
+					Requests: nil,
+					Claims:   nil,
+				},
+				ResizePolicy: nil,
+				VolumeMounts: []apiv1.VolumeMount{
+					{
+						Name:             hostVolumeName,
+						ReadOnly:         false,
+						MountPath:        hostMountPath,
+						SubPath:          "",
+						MountPropagation: nil,
+						SubPathExpr:      "",
+					},
+				},
+				VolumeDevices:            nil,
+				LivenessProbe:            nil,
+				ReadinessProbe:           nil,
+				StartupProbe:             nil,
+				Lifecycle:                nil,
+				TerminationMessagePath:   "",
+				TerminationMessagePolicy: "",
+				ImagePullPolicy:          "",
+				SecurityContext: &apiv1.SecurityContext{
+					Privileged:               &isPrivileged,
+					Capabilities:             nil,
+					SeccompProfile:           nil,
+					ProcMount:                nil,
+					ReadOnlyRootFilesystem:   nil,
+					AllowPrivilegeEscalation: nil,
+					RunAsNonRoot:             nil,
+					RunAsGroup:               nil,
+					RunAsUser:                nil,
+					SELinuxOptions:           nil,
+					WindowsOptions:           nil,
+				},
+				Stdin:     false,
+				StdinOnce: false,
+				TTY:       false,
+			},
+		},
+		[]apiv1.Volume{
+			{
+				Name:         hostVolumeName,
+				VolumeSource: manager.GetVolumeSourceForHostPath("/"), // mount the entire host filesystem in this volume
+			},
+		},
+		"",
+		"",
+		nil,
+		map[string]string{
+			"kubernetes.io/hostname": nodeName,
+		},
+		hasHostPidAccess,
+		hasHostNetworkAccess,
+	)
+	defer func() {
+		// Don't block on removing this remove directory pod because this can take a while sometimes in k8s
+		go func() {
+			removeCtx := context.Background()
+			if removeDataDirPod != nil {
+				err := manager.RemovePod(removeCtx, removeDataDirPod)
+				if err != nil {
+					logrus.Warnf("Attempted to remove pod '%v' in namespace '%v' but an error occurred:\n%v", removeDataDirPod.Name, namespace, err.Error())
+					logrus.Warn("You may have to remove this pod manually.")
+				}
+			}
+		}()
+	}()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred creating pod '%v' in namespace '%v'.", removeDataDirPodName, namespace)
+	}
+
+	removeDirSuccessExitCode := int32(0)
+	dirPathToRemoveAndEmpty := fmt.Sprintf("%v%v", hostMountPath, dirPathToRemove)
```

Comment:
Same question as above - since you know exactly the directory you want to remove, why not just mount only this?

EDIT: I think, if either one of these variables is empty, you could end up with a `rm -rf /`